### PR TITLE
SOF-752 add support for multiple Viz shows

### PR DESCRIPTION
**Pull request by ianshade**
Depends on https://github.com/tv2/tv-automation-state-timeline-resolver/pull/23

### CODE REVIEW

File: src/tv2_afvd_showstyle/getRundown.ts [hunk 1]
```diff
@@ -542,18 +542,21 @@ function getGlobalAdLibPiecesAFVD(context: IStudioUserContext, config: Blueprint
...
+								layer: GraphicLLayer.GraphicLLayerDesign,
+								content: {
+									deviceType: TSR.DeviceType.VIZMSE,
+									type: TSR.TimelineContentTypeVizMSE.ELEMENT_INTERNAL,
+									templateName: 'BG_LOADER_SC',
+									templateData: [],
+									showId: config.selectedGraphicsSetup.OvlShowId
+								}
+						  })
+						: null,
```
Comment (LindvedKrvang) [inline src/tv2_afvd_showstyle/getRundown.ts:559]: Not a fan of deliberately adding null, it forces us to do null check further down the line. 
If we extract the array of timelineObjects to a local variable, then we can just push the timelineObject if the type is Viz and do nothing if it is not.

**Review by skon-tv2 — APPROVED** [review]

File: src/tv2_afvd_showstyle/helpers/config.ts [hunk 2]
```diff
@@ -4,28 +4,45 @@ import {
...
+	return foundTableConfigGraphicsSetup
 }
 
-export function getConfig(context: IShowStyleContext): BlueprintConfig {
-	return ({ ...(context.getStudioConfig() as any), ...(context.getShowStyleConfig() as any) } as any) as BlueprintConfig
+export function parseConfig(context: ICommonContext, rawConfig: IBlueprintConfig): any {
+	const showstyleConfig = (rawConfig as unknown) as ShowStyleConfig
+	const selectedGraphicsSetup = findGraphicsSetup(context, showstyleConfig)
+	return {
+		showStyle: showstyleConfig,
```
[the remark this answers is not in the document]
Comment (ianshade) [reply src/tv2_afvd_showstyle/helpers/config.ts:41]: I see your point, but when using it, it usually looks like `config.showStyle.someProp`, which seems less cluttered than `config.showStyleConfig.someProp`

File: src/tv2-common/helpers/graphics/index.ts [hunk 3]
```diff
@@ -149,3 +149,15 @@ export function CreateGraphicBaseline(config: TV2BlueprintConfig): TSR.TSRTimeli
 		]
 	}
 }
+
+export function findShowId(config: TV2BlueprintConfig, engine: GraphicEngine) {
+	const graphicsSetup = config.selectedGraphicsSetup
+	switch (engine) {
+		case 'FULL':
```
[the remark this answers is not in the document]
Comment (ianshade) [reply src/tv2-common/helpers/graphics/index.ts:156]: Luckily the union is properly type-checked. If I remove some string or add another to the union, I'll know immediately. 
image
Screenshot 2022-03-16 at 10 11 35
IMO it boils down to the ease of reading (+1 for union) vs ease of refactoring (+1 for enums)

File: src/tv2_afvd_showstyle/helpers/config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/tv2_afvd_showstyle/helpers/config.ts:41]: Agree that "config.showStyle.someProp" looks better

**Pull request merged**